# Week3a: Metabarcoding Journal Club

Before class, please read the following journal articles in full:

Deiner K, Bik HM, Mächler E, Seymour M, Lacoursière‐Roussel A, Altermatt F, Creer S, Bista I, Lodge DM, De Vere N, Pfrender ME. (2017) Environmental DNA metabarcoding: Transforming how we survey animal and plant communities. _Molecular Ecology_, 26(21):5872-95. https://onlinelibrary.wiley.com/doi/full/10.1111/mec.14350

Abellan-Schneyder I, Matchado MS, Reitmeier S, Sommer A, Sewald Z, Baumbach J, List M, Neuhaus K. (2021) Primer, pipelines, parameters: issues in 16S rRNA gene sequencing. _mSphere_. 2021 Feb 24;6(1):10-128. https://journals.asm.org/doi/10.1128/msphere.01202-20

Perry WB, Pillay K, George P, Brennan G, Lowe A, Jones L, Holman L, Gibson T, de Vere N, Creer S. (2023) Design considerations for eDNA metabarcoding surveys. _Applied Environmental Genomics_, book chapter pp2-13. PDF: https://www.dropbox.com/scl/fi/4dm31ga36r5k96p7ecylh/Perry-et-al.-Design-considerations-for-eDNA-metabarcoding-surveys.pdf?rlkey=ar5id8vne5war1tv1xwdqu9pf&dl=0 (full book is on Google Books and has lots of other great chapters, if you're interested! https://www.google.com/books/edition/Applied_Environmental_Genomics/34XmEAAAQBAJ)

---

Slide Deck from Today's class: https://www.dropbox.com/scl/fi/zdzxu49sogvy1ufsia1ib/Week3a-Metabarcoding-JournalClub.pdf?rlkey=iq2yuq13h656lurl3uspo2ydm&dl=0

## Exercise 1: 

#### In pairs, discuss the different sources of environmental DNA that may be present in your class dataset (e.g. the data you'll be working on, or another example marine dataset you're familiar with such as TARA Oceans, the Earth Microbiome Project,etc.)
* What is the main **target** source of eDNA/eRNA for your study? (the biological community or DNA/RNA pool you are aiming to analyze)
* What is the most important source of **non-target** eDNA/eRNA in your study? How will you quantify, characterize, and assess non-target eDNA/eRNA contributions during analysis?
* Are there **any other sources** of eDNA/eRNA that are important and pertinent to your study?

Follow GDoc questions (and take notes here) for this excercise: https://docs.google.com/document/d/1HSeWThmgfKe2F9ay3mX6YhBXyJ_FIkSJcoFv5qFf_XA/edit?usp=sharing


## Exercise 2:

#### In pairs, discuss the different biases that your field collections and wet lab protocols may introduce in your class dataset.

* What is the most important source of bias that your **fieldwork or sample collection strategies** introduce into your dataset? (the biological community or DNA/RNA pool you are aiming to analyze)
* What is the most important source of bias that your **wet lab protocols and sequencing strategy** introduce into your dataset? (e.g. DNA extractions, PCR, library prep, the actual sequencing platform itself)
* What steps might you take to help you **assess (and/or mitigate) the sources of bias** you identified in your fieldwork/sampling? Wet lab and sequencing protocols?

Follow GDoc questions (and take notes here) for this excercise: https://docs.google.com/document/d/1HijY66D_IFm7MMntNAhJ3tU4PrCgwRlI1WBFWwf3YAA/edit?usp=sharing

---

#### Class Plan

* 1:15 - 1:30: Lecture on sequencing technologies
* 1:30 - 1:35: Q&A
* 1:35 - 1:45: What is eDNA
* 1:45 - 1:55: Exercise #1
* 1:55 - 1:15: Sources of eDNA
* 1:15 - 1:25: Exercise #2
* 1:25 - 1:35: Group discussion and Wrap-up

